google-cloud-storage/lib/google/cloud/storage/file.rb

```diff
@@ -639,6 +639,9 @@ def rotate encryption_key: nil, new_encryption_key: nil
         # Permanently deletes the file.
         #
         # @return [Boolean] Returns `true` if the file was deleted.
+        # @param [Boolean,Integer] generation Select a specific revision of the
+        #   file to delete. When `true` it will use the object's generation. The
```

Comment:
If you call `Bucket#file` with the `generation` argument you can get an object with that specific generation version. It does not have to be the latest generation version. But the current behavior is to always delete the latest version, so this is an attempt to preserve the existing behavior.